browser/src/control/Control.FormulaBarJSDialog.js

```diff
@@ -45,7 +45,6 @@ class FormulaBar {
 		this.map.off('jsdialogaction', this.onJSAction, this);
 
 		this.map.off('doclayerinit', this.onDocLayerInit, this);
-		this.map.off('updatepermission', this.onUpdatePermission, this);
```

Comment:
we should have `app.events.off`

Reply:
When is this function called? I checked it now and couldn't trigger it. Maybe wen can remove this function.

Reply:
I think it is triggered when we switch edit -> readonly mode for example, easy to check on tablet, or when disconnected the server or connected again.

Reply:
I couldn't hit the codepath unfortunately.

Reply:
please see `browser/src/control/Permission.js` `_enterEditMode` and `_enterReadOnlyMode`

Reply:
I can't hit there with/without an integrator on mobile view. Let me check this later again.